src/primitive_core.cc

```diff
@@ -1601,17 +1601,16 @@ PRIMITIVE(array_at_put) {
 // Allocates a new array and copies old_length elements from the old array into
 // the new one.
 PRIMITIVE(array_expand) {
-  ARGS(Array, old, word, old_length, word, length);
+  ARGS(Array, old, word, old_length, word, length, Object, filler);
   if (length == 0) return process->program()->empty_array();
   if (length < 0) OUT_OF_BOUNDS;
-  if (length > Array::max_length_in_process()) OUT_OF_RANGE;
-  if (old_length < 0 || old_length > old->length() || old_length > length) OUT_OF_RANGE;
-  Object* result = process->object_heap()->allocate_array(length, Smi::from(0));
+  if (length > Array::ARRAYLET_SIZE) OUT_OF_RANGE;
```

Comment:
ARRAYLET_SIZE is 500 right now. Is that a good number? Can we make it much larger on 64-bit platforms where the page size is bigger?